lib/portage/gpkg.py

```diff
@@ -0,0 +1,967 @@
+# Copyright 2001-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+import tarfile
+import io
+from datetime import datetime
+import warnings
+import threading
+import subprocess
+import errno
+import pwd
+import grp
+import stat
+import sys
+
+from portage import os
+from portage import shutil
+from portage import normalize_path
+from portage import _encodings
+from portage import _unicode_decode
+from portage import _unicode_encode
+from portage.exception import InvalidData, FileNotFound, \
+	IncorrectParameter, InvalidBinaryPackageFormat, \
+	InvalidCompressionMethod, CompressorNotFound, \
+	CompressorOperationFailed
+from portage.output import colorize
+from portage.util._urlopen import urlopen
+from portage.util import writemsg
+from portage.util import ensure_dirs, shlex_split, varexpand
+from portage.util.compression_probe import _compressors
+from portage.process import find_binary
+
+
+class tar_stream_writer(object):
+	"""
```

Comment:
I think it would be nice to give a quick example of its usage.

Reply:
TODO